zipline/utils/tradingcalendar.py

```diff
@@ -12,393 +11,250 @@
-        dtstart=start,
-        # valid 1993-2007
-        until=min(end, datetime(2007, 12, 31, tzinfo=pytz.utc))
-    )
-    early_close_rules.append(friday_after_christmas)
-
-    day_before_independence_day = rrule.rrule(
-        rrule.MONTHLY,
-        bymonth=7,
-        bymonthday=3,
-        byweekday=(rrule.MO, rrule.TU, rrule.TH),
-        cache=True,
-        dtstart=start,
-        until=end
-    )
-    early_close_rules.append(day_before_independence_day)
-
-    day_after_independence_day = rrule.rrule(
-        rrule.MONTHLY,
-        bymonth=7,
-        bymonthday=5,
-        byweekday=rrule.FR,
-        cache=True,
-        dtstart=start,
-        # starting in 2013: wednesday before independence day
-        until=min(end, datetime(2012, 12, 31, tzinfo=pytz.utc))
-    )
-    early_close_rules.append(day_after_independence_day)
-
-    wednesday_before_independence_day = rrule.rrule(
-        rrule.MONTHLY,
-        bymonth=7,
-        bymonthday=3,
-        byweekday=rrule.WE,
-        cache=True,
-        # starting in 2013
-        dtstart=max(start, datetime(2013, 1, 1, tzinfo=pytz.utc)),
-        until=max(end, datetime(2013, 1, 1, tzinfo=pytz.utc))
-    )
-    early_close_rules.append(wednesday_before_independence_day)
-
-    early_close_ruleset = rrule.rruleset()
-
-    for rule in early_close_rules:
-        early_close_ruleset.rrule(rule)
-    early_closes = early_close_ruleset.between(start, end, inc=True)
-
-    # Misc early closings from NYSE listing.
-    # http://www.nyse.com/pdfs/closings.pdf
-    #
-    # New Year's Eve
-    nye_1999 = datetime(1999, 12, 31, tzinfo=pytz.utc)
-    if start <= nye_1999 and nye_1999 <= end:
-        early_closes.append(nye_1999)
-
-    early_closes.sort()
-    return pd.DatetimeIndex(early_closes)
 
-early_closes = get_early_closes(start, end)
 
+def days_at_time(days, t, tz):
+    """
+    Shift an index of days to time t, interpreted in tz.
 
-def get_open_and_close(day, early_closes):
-    market_open = pd.Timestamp(
-        datetime(
-            year=day.year,
-            month=day.month,
-            day=day.day,
-            hour=9,
-            minute=31),
-        tz='US/Eastern').tz_convert('UTC')
-    # 1 PM if early close, 4 PM otherwise
-    close_hour = 13 if day in early_closes else 16
-    market_close = pd.Timestamp(
-        datetime(
-            year=day.year,
-            month=day.month,
-            day=day.day,
-            hour=close_hour),
-        tz='US/Eastern').tz_convert('UTC')
+    Overwrites any existing tz info on the input.
+    """
+    days = DatetimeIndex(days).tz_localize(None).tz_localize(tz)
+    return days.shift(
+        1, freq=DateOffset(hour=t.hour, minute=t.minute, second=t.second)
+    ).tz_convert('UTC')
 
-    return market_open, market_close
 
+def holidays_at_time(calendar, start, end, time, tz):
+    return days_at_time(
+        calendar.holidays(
+            # Workaround for https://github.com/pydata/pandas/issues/9825.
```

Comment:
And looks like this can be removed with next pandas release, because of your bug fix?